cmake/dependencies/curlpp.cmake

```diff
@@ -7,6 +7,51 @@ set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/modules/ ${CMAKE_MODULE_PATH})
 include(FetchContent)
 cmake_policy(SET CMP0079 NEW)
 
+# Requires curl
+find_package(CURL QUIET)
```

Comment:
You're attempting to find/install 2 dependencies here `libcurl` and `curlpp`, I expect @ptheywood will want them to be placed in separate dependency modules for reasons of clarity (and he wouldn't be wrong).

Reply:
`CURL` is only required as a dependency of `curlpp`, but curlpp's CMake doesn't fetch it if not found (most of the code is there, but there's a `REQUIRED` which blocks it).

It could be separated out, but given we *only* need CURL for curlpp, there's an argument for them to be in the same file (e.g. an CMake option maskign out curlpp fetching would also implicitly mask out requiring / fetching curl.)

Reply:
Idea of libCurl as a dependency has been dropped